cmd/publicize/server.go

```diff
@@ -0,0 +1,188 @@
+	githubTokenGenerator func() []byte
+
+	gitName  string
+	gitEmail string
+
+	bonds map[string]string
+
+	ghc githubClient
+	gc  git.ClientFactory
+
+	remoteHost string
+
+	dry bool
+}
+
+func (s *server) handleIssueComment(l *logrus.Entry, ic github.IssueCommentEvent) {
+	if !publicizeRe.MatchString(ic.Comment.Body) {
+		return
+	}
+
+	org := ic.Repo.Owner.Login
+	repo := ic.Repo.Name
+	num := ic.Issue.Number
+	commentAuthor := ic.Comment.User.Login
+
+	logger := logrus.WithFields(logrus.Fields{
+		github.OrgLogField:  org,
+		github.RepoLogField: repo,
+		github.PrLogField:   num,
+	})
+
+	if !ic.Issue.IsPullRequest() {
+		s.createComment(ic, fmt.Sprintf("Publicize plugin can only be used in pull requests"), logger)
+		return
+	}
+
+	// Only org members should be able to publicize a pull request.
+	ok, err := s.ghc.IsMember(org, commentAuthor)
+	if err != nil {
+		return
+	}
+	if !ok {
+		s.createComment(ic, fmt.Sprintf("only [%s](https://github.com/orgs/%s/people) org members may request a publicize", org, org), logger)
+		return
+	}
+	logger.Info("Publicize of PR has been requested.")
+
+	pr, err := s.ghc.GetPullRequest(org, repo, num)
+	if err != nil {
+		logger.WithError(err).Warn("couldn't get pull request")
+		return
+	}
+	baseBranch := pr.Base.Ref
+
+	if !pr.Merged {
+		s.createComment(ic, "cannot publicize an unmerged pull request", logger)
+		return
+	}
+
+	logger.Info("Searching bond...")
+	destOrgRepo, ok := s.bonds[fmt.Sprintf("%s/%s", org, repo)]
+	if !ok {
+		logger.Warn("There is no public repository configured for the current repository.")
+		s.createComment(ic, "cannot publicize because there is no bond for the current pull request", logger)
+		return
+	}
+
+	destOrg := strings.Split(destOrgRepo, "/")[0]
+	destRepo := strings.Split(destOrgRepo, "/")[1]
+	sourceRemoteResolver := func() (string, error) {
+		u, err := url.Parse(s.remoteHost)
+		if err != nil {
+			return "", err
+		}
+
+		// If the host isn't a url but its an actual path, return the full path with org and repo included.
+		// This is usefull for testing as well.
+		if u.Host == "" {
+			return fmt.Sprintf("%s/%s/%s", s.remoteHost, org, repo), nil
+		}
+
+		u.Path = path.Join(org, repo)
+		if s.githubTokenGenerator != nil {
+			u.User = url.User(string(s.githubTokenGenerator()))
+		}
+		return u.String(), nil
+	}
+
+	logger.Infof("Trying to merge the PR to destination: %s/%s@%s", destOrg, destRepo, baseBranch)
+	mergeMsg := fmt.Sprintf("merge history from %s/%s", org, repo)
+	if err := s.mergeAndPushToRemote(destOrg, destRepo, sourceRemoteResolver, baseBranch, mergeMsg, s.dry); err != nil {
+		logger.WithError(err).Warnf("couldn't merge the pull request and push to the destination: %v", err)
+		s.createComment(ic, fmt.Sprintf("Publicize failed with error: %v", err), logger)
+		return
+	}
+
+	destOrgRepoLink := fmt.Sprintf("https://github.com/%s/tree/%s", destOrgRepo, baseBranch)
+	s.createComment(ic, fmt.Sprintf("Publicize to [%s/%s@%s](%s) was successful",
```

Comment:
Maybe reword:

```
A pull request at [%s/%s#%d]() was created in the upstream repository to publish this work.
```

Reply:
Well, actually it is not creating any pull request, but it direct merges it.

Reply:
Outstanding -- if's a commit, then 

```
A merge commit [%s/%s@%s]() was created in the upstream repository to publish this work.
```